### PR TITLE
Upgraded SearchPreference library to version 2.0.0

### LINKED ISSUE #4509: Update SearchPreference to latest version
@ByteHamster: Would you be up for updating K-9 Mail to the latest SearchPreference release?

### PR DESCRIPTION
**Pull request by ByteHamster**
Closes #4509

### CODE REVIEW

File: app/ui/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -786,6 +786,7 @@ Please submit bug reports, contribute new features and ask questions at
     <string name="settings_message_theme_label">Message view theme</string>
     <string name="settings_compose_theme_label">Composer theme</string>
     <string name="settings_language_label">Language</string>
+    <string name="preference_search_no_results">No settings found</string>
```
Comment (cketti) [inline app/ui/src/main/res/values/strings.xml]: Is this overriding a string provided by the library? If so, please put it into a separate section, using a header similar to what we use here (include the library name): 
https://github.com/k9mail/k-9/blob/de1612fb16f51a074cfa12f5f12874cdc93d717c/app/ui/src/main/res/values/strings.xml#L66

Comment (ByteHamster) [reply app/ui/src/main/res/values/strings.xml]: Interesting. I did not know that you can overwrite library strings. The string is not directly overwriting the library string. It uses `setTextNoResults(...)` of the library.

Comment (cketti) [reply app/ui/src/main/res/values/strings.xml]: My bad. I couldn't find a usage of the string. But it's clearly there :)

**Comment by cketti** [comment]
Thanks :+1:

I noticed the app is crashing on orientation change when search results are displayed. It doesn't seem to be related to the update, though.
Can you please look into this when you find some time. I created issue #4519.

**Pull request merged**